NumPy_exercises.py

```diff
@@ -0,0 +1,286 @@
+
+
+def show_the_sine_wave_of_a_given_frequency(frequency: int) -> None:
+    """
+    Show the sine wave of a given frequency between -pi and pi.
+    Args:
+        frequency (int): Frequency of the sine wave.
+    """
+    # create a vector of the given frequency
+    x = np.linspace(np.pi * -1, np.pi, frequency)
+    y = np.sin(x)
+    # plot the sine wave
+    plt.plot(x, y)
+    plt.show()
+
+
+def swap_rows_of_a_matrix(matrix: np.ndarray, row_1: int, row_2: int) -> np.ndarray:
+    """
+    Swap two rows of a matrix.
+    Args:
+        matrix (np.ndarray): Matrix to swap rows.
+        row_1 (int): First row to swap.
+        row_2 (int): Second row to swap.
+
+    Returns:
+        np.ndarray: Matrix with swapped rows.
+    """
+    # create a copy of the matrix
+    matrix_copy = matrix.copy()
+    # swap the rows
+    matrix[row_1, :], matrix[row_2, :] = matrix_copy[row_2, :], matrix_copy[row_1, :]
+    return matrix
+
+
+def mapping_array_by_given_number(array: np.ndarray, number: int = 5,
+                                  switch_smaller_to: int = -1,
+                                  switch_larger_to: int = 1,
+                                  switch_equal_to: int = 0) -> np.ndarray:
+    """
+    Map an array by a given number.
+    Args:
+        array (np.ndarray): Array to map.
+        number (int, optional): Number to map the array by. Defaults to 5.
+        switch_smaller_to (int, optional): Value to change the smaller numbers to. Defaults to 0.
+        switch_larger_to (int, optional): Value to change the bigger numbers to. Defaults to 1.
+        switch_equal_to (int, optional): Value to change the equal numbers to. Defaults to 0.
+
+    Returns:
+        np.ndarray: Mapped array.
+    """
+    # create a copy of the array
+    array_copy = array.copy()
+    # map the array
+    array_copy[array_copy < number] = switch_smaller_to
+    array_copy[array_copy > number] = switch_larger_to
+    array_copy[array_copy == number] = switch_equal_to
+    return array_copy
+
+
+def combine_one_and_two_dimensional_array_together(array_1: np.ndarray, array_2: np.ndarray) -> None:
+    """
+        Combine one and two dimensional array together.
+    Args:
+        array_1 (np.ndarray): First array to combine.
+        array_2 (np.ndarray): Second array to combine.
+    """
+    print("array_1:\n", array_1)
+    print("array_2:\n", array_2)
+    print("combined array:")
+    # combine the arrays
+    for first, second in np.nditer([array_1, array_2]):
+        print(f" {first} : {second}")
+
+
+def get_days_in_month(date: str) -> int:
+    """
+    Get the number of days in a given month.
+    Args:
+        date (str): Date to get the number of days in format: mm-yyyy.
+        
+    Returns:
+        int: Number of days in the month.
+    """
+    # get the month and year from the date
+    month, year = date.split("-")
+    # get the number of days in the month
+    return calendar.monthrange(int(year), int(month))[1]
```

Comment:
Notice numpy has a datetime functionality that could be used here as well-
https://numpy.org/doc/stable/reference/arrays.datetime.html